logger/logger.go

```diff
@@ -83,36 +87,89 @@ func (l *Logger) SetServiceName(name string) {
 	l.ServiceName = name
 }
 
-func (l *Logger) Info(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("info", params)
+func (l *Logger) Context(id LoggerEventID, extra ...ExtraData) *LoggerEvent {
+	var extraData *ExtraData
+
+	if len(extra) > 0 {
+		extraData = &extra[0]
+	}
+
+	return &LoggerEvent{
+		logLevel: "info",
+		logger:   l,
+		id:       id,
+		extra:    extraData,
+	}
+}
+
+func (l *Logger) Info() *LoggerEvent {
+	return l.getLoggerEvent("info")
 }
 
-func (l *Logger) Debug(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("debug", params)
+func (l *Logger) Debug() *LoggerEvent {
+	return l.getLoggerEvent("debug")
 }
 
-func (l *Logger) Warn(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("warn", params)
+func (l *Logger) Warn() *LoggerEvent {
+	return l.getLoggerEvent("warn")
 }
 
-func (l *Logger) Error(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("error", params)
+func (l *Logger) Error() *LoggerEvent {
+	return l.getLoggerEvent("error")
+}
+
+func (event *LoggerEvent) Info() *LoggerEvent {
+	params := []interface{}{event.id, event.extra}
+	return event.logger.getLoggerEvent("info", params...)
```

Comment:
Why not just `return event.logger.getLoggerEvent("info", event.id, event.extra)`?